src/Microsoft.ML.Data/Prediction/Calibrator.cs

```diff
@@ -2070,6 +2075,195 @@ private float FindValue(float score)
+            var maxesLengthVar = ctx.AddInitializer(Maxes.Length, "MaxesLength");
+            var minToReturnVar = ctx.AddInitializer((float)1e-15, "MinToReturn");
+            var maxToReturnVar = ctx.AddInitializer((float)(1 - 1e-15), "MaxToReturn");
+            var minsVar = ctx.AddInitializer(Mins, new long[] { Mins.Length, 1 }, "Mins");
+            var maxesVar = ctx.AddInitializer(Maxes, new long[] { Maxes.Length, 1 }, "Maxes");
+            var valuesVar = ctx.AddInitializer(Values, new long[] { Values.Length, 1 }, "Values");
+            var zeroVar = ctx.AddInitializer(0, "Zero");
+            var oneVar = ctx.AddInitializer(1, "One");
+
+            //The isotonic regression optimization problem is defined by:
+            //  min(sum(w_i, (y[i] - y_[j])^2))
+            //  subject to y_[i] <= y_[j] whenever X[i] <= X[j] (non-decreasing)
+            //  and min(y_) = y_min, max(y_) = y_max
+            //  where:
+            //    *y[i] are inputs(real numbers)
+            //    *y_[i] are fitted
+            //    *X specifies the order.If X is non-decreasing then y_ is non-decreasing.
+            //    *w[i] are optional strictly positive weights(default to 1.0)
+
+            // Goal: Given output, calculate prob
+
+            // --- STEP 1: implement if-then-else logic for (p = Mins.Length): ------------------------------------
+            //     If p == 0, Return 0
+            //     If score < Mins[0], Return prob = Values[0]
+            //     If score > Maxes[p-1], Return prob = Values[p-1]
+            //     Else, continue
+
+            // Get Mins[0]
+            string opType = "GatherElements";
+            var minsZeroOutput = ctx.AddIntermediateVariable(NumberDataViewType.Single, "minsZeroOutput");
+            OnnxNode node = ctx.CreateNode(opType, new[] { minsVar, zeroVar }, new[] { minsZeroOutput }, ctx.GetNodeName(opType), "");
+
+            // Get Maxes[p-1] = Maxes[Mins.Length-1]
+            opType = "GatherElements";
+            var maxesPMinusOneOutput = ctx.AddIntermediateVariable(NumberDataViewType.Single, "maxesPMinusOneOutput");
+            node = ctx.CreateNode(opType, new[] { maxesVar, minsLengthMinusOneVar }, new[] { maxesPMinusOneOutput }, ctx.GetNodeName(opType), "");
+
+            // Get Values[0]
+            opType = "GatherElements";
+            var valuesZeroOutput = ctx.AddIntermediateVariable(NumberDataViewType.Single, "valuesZeroOutput");
+            node = ctx.CreateNode(opType, new[] { valuesVar, zeroVar }, new[] { valuesZeroOutput }, ctx.GetNodeName(opType), "");
+
+            // Get Values[p-1] = Values[Mins.Length-1]
+            opType = "GatherElements";
+            var valuesPMinusOneOutput = ctx.AddIntermediateVariable(NumberDataViewType.Single, "valuesPMinusOneOutput");
+            node = ctx.CreateNode(opType, new[] { valuesVar, minsLengthMinusOneVar }, new[] { valuesPMinusOneOutput }, ctx.GetNodeName(opType), "");
+
```

Comment:
Aren't `Mins[0]`, `Maxes[p-1]` constant for the model? Can use initializers instead of using GatherElements?